wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/permissions/OrganizationResourceDistributionServicePermissionsFilter.java

```diff
@@ -0,0 +1,92 @@
+import org.eclipse.che.commons.subject.Subject;
+import org.eclipse.che.everrest.CheMethodInvokerFilter;
+import org.everrest.core.Filter;
+import org.everrest.core.resource.GenericResourceMethod;
+
+import javax.inject.Inject;
+import javax.ws.rs.Path;
+
+import static com.codenvy.organization.api.permissions.OrganizationDomain.DOMAIN_ID;
+import static com.codenvy.organization.api.permissions.OrganizationPermissionsFilter.MANAGE_ORGANIZATIONS_ACTION;
+
+/**
+ * Restricts access to methods of {@link OrganizationResourcesDistributionService} by users' permissions.
+ *
+ * <p>Filter contains rules for protecting of all methods of {@link OrganizationResourcesDistributionService}.<br>
+ * In case when requested method is unknown filter throws {@link ForbiddenException}
+ *
+ * @author Sergii Leschenko
+ */
+@Filter
+@Path("/organization/resource{path:(/.*)?}")
+public class OrganizationResourceDistributionServicePermissionsFilter extends CheMethodInvokerFilter {
+    static final String DISTRIBUTE_RESOURCES_METHOD      = "distribute";
+    static final String GET_DISTRIBUTED_RESOURCES_METHOD = "getDistributedResources";
+    static final String RESET_DISTRIBUTED_RESOURCES      = "reset";
+
+    @Inject
+    private OrganizationManager organizationManager;
+
+    @Override
+    protected void filter(GenericResourceMethod genericMethodResource, Object[] arguments) throws ApiException {
+        final String methodName = genericMethodResource.getMethod().getName();
+
+        final Subject currentSubject = EnvironmentContext.getCurrent().getSubject();
+        String organizationId;
+        switch (methodName) {
+            case DISTRIBUTE_RESOURCES_METHOD:
+            case RESET_DISTRIBUTED_RESOURCES:
+                //we should check permissions on parent organization level
+                String suborganizationId = (String)arguments[0];
+                final String parentId = organizationManager.getById(suborganizationId).getParent();
+                if (parentId == null) {
+                    // requested organization is root so manager should throw exception
+                    return;
+                }
+                organizationId = parentId;
+                break;
+
+            case GET_DISTRIBUTED_RESOURCES_METHOD:
+                organizationId = (String)arguments[0];
+                // get organization to ensure that organization exists
+                organizationManager.getById(organizationId);
+                if (currentSubject.hasPermission(SystemDomain.DOMAIN_ID, null, MANAGE_ORGANIZATIONS_ACTION)) {
+                    //user is able to see information about all organizations
+                    return;
+                }
+                break;
+
+            default:
+                throw new ForbiddenException("The user does not have permission to perform this operation");
+        }
+
+
+        if (!currentSubject.hasPermission(DOMAIN_ID, organizationId, OrganizationDomain.MANAGE_RESOURCES)) {
```

Comment:
Also you use `MANAGE_ORGANIZATION_ACTION` as static import 
```
import static com.codenvy.organization.api.permissions.OrganizationPermissionsFilter.MANAGE_ORGANIZATIONS_ACTION;
```
but you don't do it with `OrganizationDomain.MANAGE_RESOURCES` is there any regularity i don't see?